### DISCUSSION TITLE
#asking-for-help : How to use mdn markdown question

### DISCUSSION THREAD

**Discussion by mineclover** [mdn/mdn-community#324] [Content] [answered]
I use mdn markdown

1. in local folder
contents ,
yari,
translated-content ,
markdown,
and install

2. Activate translated-content in the yari folder and create html with dev and build

Afterwards in markdown as per the documentation
contents
When using translated-content
No HTML documents matched the search query! Exiting.

an error occurs
It is the same even if you set the paths with build in yari.

2. Not sure about the next step
Do you know how to use it properly?

**Comment by yin1999** [comment, marked as answer]
Hi @mineclover. If you only want to see the rendered HTML content in browser. I think it could be more easier:

- clone mdn/content and mdn/translated-content (optionally, only when you are interested in the translated documents)
- go to the local `mdn/content` folder, run command `yarn install`
- optionally create a `.env` file in `mdn/content` repo, and set an variable (as the [step 3](https://github.com/mdn/translated-content#content-repo-setup))
- run `yarn start` in content folder, and open the URL `http://localhost:5042/` in browser.

And if you want to build markdown document into html files. Just use the command `yarn build args...` (run `yarn build --help` to see args that needs to be set)

Only if you are interested in developing the platform ([mdn/yari](https://github.com/mdn/yari)), you could use `yarn dev` to run a dynamic SSR mode (would not build out static HTML files in file system), or run `yarn build` to build out static HTML files (you could follow the commands that used for [prod build](https://github.com/mdn/yari/blob/071153f920d02cae78876eafcb04484bd2039622/.github/workflows/prod-build.yml#L187-L217) to get this done).

If you have any other question, feel free to append this thread.

**Reply by mineclover** [reply]
Ah, the original "content" was difficult, but thanks to you, I did it.

I'm working on a feature to create markdown.

../content/files/ returns "No HTML documents matched the search query! Exiting."
../content/files/build is
Error: ENOENT: no such file or directory, lstat 'P:\Dev\MDN\content\files\build'

**Reply by yin1999** [reply]
Do you mean converting HTML documents to markdown?

We are now 100% markdown for both mdn/content and mdn/translated-content. So if you run the command trying to convert html files to markdown, it will exit immediately.

for `content/files/build` (I guess you are trying to convert the static HTML file built by yari), it could not be done by mdn/markdown. So you could stop trying this. We just use this tool to converting raw HTML file we have used before in mdn/content and mdn/translated-content. See also: https://github.com/mdn/translated-content/discussions/1236

**Reply by mineclover** [reply]
yes .. So I know that mdn/markdown is a separate html2md module
(This is also written in the package.json script)
So I tried to run mdn/markdown normally, but it didn't work.

I looked at the mdn/markdown repo
Looking at the commit log, there was an update recently. Is this a module that is no longer used??

**Reply by yin1999** [reply]
First of all, you should make sure that, HTML file existed in mdn/content or mdn/translated-content. For now, they are both in fully markdown format.

If you want to try this tool, please checkout a commit before any markdown convertion commit (like this one: mdn/content#7092). And then run convertion command on the specific folder.

Or you could create a `index.html` file in a new folder, and run convertion command to convert the html file into markdown.

**Reply by hochan222** [reply]
If you want to know about the markdown conversion history, you can refer to the links below :)

- https://hacks.mozilla.org/2022/09/the-100-percent-markdown-expedition/
- https://github.com/mdn/translated-content/issues/7486

**Reply by mineclover** [reply]
That's not what I want.

I will understand that mdn/markdown is used to change the HTML documents that are still left.
